stdlib/domain.ml

```diff
@@ -37,7 +37,7 @@ type 'a state =
 | Finished of ('a, exn) result
 | Joined
 
-type 'a t = {
+type !'a t = {
```

Comment:
I don't think this annotation here is necessary or very helpful -- all nominal types are injective, so this record type will directly be inferred as injective.

Reply:
I did not add it in the beginning but the testing with the native compiler failed. I could try again to see what's going on.

Reply:
It seems that I have to `make clean` in order to properly test the native compiler.